google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoder.java

```diff
@@ -0,0 +1,399 @@
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Encoder for TIME and DATETIME values, according to civil_time encoding.
+ *
+ * <p>The valid range and number of bits required by each date/time field is as the following:
+ *
+ * <table>
+ *   <tr> <th> Field  </th> <th> Range          </th> <th> #Bits </th> </tr>
+ *   <tr> <td> Year   </td> <td> [1, 9999]      </td> <td> 14    </td> </tr>
+ *   <tr> <td> Month  </td> <td> [1, 12]        </td> <td> 4     </td> </tr>
+ *   <tr> <td> Day    </td> <td> [1, 31]        </td> <td> 5     </td> </tr>
+ *   <tr> <td> Hour   </td> <td> [0, 23]        </td> <td> 5     </td> </tr>
+ *   <tr> <td> Minute </td> <td> [0, 59]        </td> <td> 6     </td> </tr>
+ *   <tr> <td> Second </td> <td> [0, 59]*       </td> <td> 6     </td> </tr>
+ *   <tr> <td> Micros </td> <td> [0, 999999]    </td> <td> 20    </td> </tr>
+ *   <tr> <td> Nanos  </td> <td> [0, 999999999] </td> <td> 30    </td> </tr>
+ * </table>
+ *
+ * <p>* Leap second is not supported.
+ *
+ * <p>When encoding the TIME or DATETIME into a bit field, larger date/time field is on the more
+ * significant side.
+ */
+public final class CivilTimeEncoder {
+  private static final int NANO_LENGTH = 30;
+  private static final int MICRO_LENGTH = 20;
+  // private static final int SECOND_LENGTH = 10;
+
+  private static final int NANO_SHIFT = 0;
+  private static final int MICRO_SHIFT = 0;
+  private static final int SECOND_SHIFT = 0;
+  private static final int MINUTE_SHIFT = 6;
+  private static final int HOUR_SHIFT = 12;
+  private static final int DAY_SHIFT = 17;
+  private static final int MONTH_SHIFT = 22;
+  private static final int YEAR_SHIFT = 26;
+
+  private static final long NANO_MASK = 0x3FFFFFFFL;
+  private static final long MICRO_MASK = 0xFFFFFL;
+  private static final long SECOND_MASK = 0x3FL;
+  private static final long MINUTE_MASK = 0xFC0L;
+  private static final long HOUR_MASK = 0x1F000L;
+  private static final long DAY_MASK = 0x3E0000L;
+  private static final long MONTH_MASK = 0x3C00000L;
+  private static final long YEAR_MASK = 0xFFFC000000L;
+
+  private static final long TIME_SECONDS_MASK = 0x1FFFFL;
+  private static final long TIME_MICROS_MASK = 0x1FFFFFFFFFL;
+  private static final long TIME_NANOS_MASK = 0x7FFFFFFFFFFFL;
+  private static final long DATETIME_SECONDS_MASK = 0xFFFFFFFFFFL;
+  private static final long DATETIME_MICROS_MASK = 0xFFFFFFFFFFFFFFFL;
+
+  /**
+   * Encodes {@code time} as a 4-byte integer with seconds precision.
+   *
+   * <p>Encoding is as the following:
+   *
+   * <pre>
+   *      3         2         1
+   * MSB 10987654321098765432109876543210 LSB
+   *                    | H ||  M ||  S |
+   * </pre>
+   *
+   * @see #decodePacked32TimeSeconds(int)
+   */
+  @SuppressWarnings("GoodTime-ApiWithNumericTimeUnit")
+  public static int encodePacked32TimeSeconds(java.time.LocalTime time) {
```

Comment:
We don't need this.

Reply:
Sounds good, will remove

Reply:
Is actually used by other functions, will make private